dev/EnvironmentManager/ChangeTracker/EnvironmentVariableChangeTrackerHelper.h

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+
+#include <windows.h>
+#include <memory>
+#include <wil/result.h>
+#include <appmodel.h>
+#include <winrt/Windows.Management.Deployment.h>
+#include <Microsoft.Windows.System.EnvironmentManager.h>
+#include <AppModel.Identity.h>
+#include <Security.IntegrityLevel.h>
+#include <Microsoft.Windows.System.EnvironmentManager.Insights.h>
+#include <WindowsAppRuntime.SelfContained.h>
+
+
+using namespace winrt::Microsoft::Windows::System::implementation;
+
+static bool DoesChangeTrackingKeyExist()
+{
+    const auto subKey = std::filesystem::path{ LR"(Software\Microsoft\AppModel\RegistryWriteVirtualization\ExcludedKeys\HKEY_CURRENT_USER/Software/ChangeTracker)" };
+    wil::unique_hkey keyToChangeTracker{};
+    LSTATUS openResult{ RegOpenKeyEx(HKEY_LOCAL_MACHINE, subKey.c_str(), 0, KEY_READ | KEY_WOW64_64KEY, keyToChangeTracker.put()) };
+
+    if (FAILED_WIN32(openResult))
+    {
+        EnvironmentManagerInsights::TraceLoggingInfo("Could not find the change tracking HKLM key.  Changes will not be tracked");
+        LOG_IF_WIN32_ERROR(openResult);
```

Comment:
What's the difference between FAILED_WIN32() and LOG_IF_WIN32_ERROR?

If you get here isn't openResult guaranteed to be a WIN32_ERROR? So this should be LOG_WIN32_ERROR(openResult)?

Of course why both? Why not

```
const HRESULT hr{ LOG_IF_WIN32_ERROR(RegOpenKeyEx(...) };
if (FAILED(hr))
{
    EnvironmentManagerInsights::TraceLoggingInfo(...)
    return false;
}

Reply:
Yes that.  I wanted to log the WIN32 if it was an error and change the return type depending on the return code.

This was the way I thought to do it.

However, your way is much better.